aws/resource_aws_eks_cluster.go

```diff
@@ -113,6 +114,28 @@ func resourceAwsEksCluster() *schema.Resource {
 					},
 				},
 			},
+
+			"kubernetes_network_config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"service_ipv4_cidr": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+							ValidateFunc: validation.All(
+								validation.IsCIDRNetwork(12, 24),
+								validation.StringMatch(regexp.MustCompile(`^(10|172\.(1[6-9]|2[0-9]|3[0-1])|192\.168)\..*`), "must be within 10.0.0.0/8, 172.16.0.0.0/12, or 192.168.0.0/16"),
```

Comment:
Please add the 100.64.0.0/10 range here too as per [How do I use multiple CIDR ranges with Amazon EKS?](https://aws.amazon.com/premiumsupport/knowledge-center/eks-multiple-cidr-ranges/) and [Amazon EKS now supports additional VPC CIDR blocks](https://aws.amazon.com/about-aws/whats-new/2018/10/amazon-eks-now-supports-additional-vpc-cidr-blocks/).

Sounds like the 198.19.0.0/16 range is valid in this scenario too but I'm not familiar with that and wouldn't want any ambiguity around it holding up this PR.

Reply:
thanks for bringing this up @etoews! looks like even though the doc suggests it for use in associating the range to a VPC, we're unfortunately blocked by the AWS API's RFC1918 validation e.g. i gave it a shot locally (removing the validation funcs in place for testing) and passing the CIDR `100.64.0.0/16` (since the netmask size must be between 12 and 24 inclusive):
```
 Error: error creating EKS Cluster (tf-acc-test-blljj): InvalidParameterException: Invalid Cluster Service Ipv4 CIDR for the cluster tf-acc-test-blljj, must be within RFC1918 range
        {
          RespMetadata: {
            StatusCode: 400,
            RequestID: "dcc64171-1ffa-4388-a1e1-e05f4d6cb1d0"
          },
          ClusterName: "tf-acc-test-blljj",
          Message_: "Invalid Cluster Service Ipv4 CIDR for the cluster tf-acc-test-blljj, must be within RFC1918 range"
        }
```

Reply:
and similar response when passing in `198.19.0.0/16`

Reply:
Yep. I was thinking about another use for those ranges. My misunderstanding. The 100.64.0.0/10 and 198.19.0.0/16 ranges are for pod IP ranges, whereas this PR is for service IP ranges. Apologies for the noise!

Reply:
no worries at all @etoews!